server/modules/nodeDef/manager/nodeDefManager.js

```diff
@@ -53,6 +55,49 @@ const updateNodeDefProps = async (user, surveyId, nodeDefUuid, props, propsAdvan
     return nodeDef
   })
 
+const deleteNodeDefValidationMessagesLabels = async (surveyId, deletedLanguages, client = db) => {
+  await client.tx(async tx => {
```

Comment:
I'm confident this huge method can be replaced by a few lines of an SQL queries, within publishProps method. Let's try tomorrow